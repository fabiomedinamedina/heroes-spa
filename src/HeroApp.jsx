import { UserProvider } from './auth/context/UserProvider';
import { AppRouter } from './router/AppRouter';

export const HeroApp = () => {
  return (
    <>
      <UserProvider>
        <AppRouter />
      </UserProvider>
    </>
  );
};
