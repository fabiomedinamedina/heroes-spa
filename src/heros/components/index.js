export * from './HeroItem';
export * from './HerosList';
export * from './TitlePublisher';