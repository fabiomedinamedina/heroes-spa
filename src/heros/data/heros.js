export const heroes = [
  {
    id: "dc-batman",
    superhero: "Batman",
    publisher: "DC Comics",
    alter_ego: "Bruce Wayne",
    first_appearance: "Detective Comics #27",
    characters: "Bruce Wayne",
    description: "Batman (conocido inicialmente como Bat-Man y en español como el Hombre Murciélago) es un personaje de cómic creado por los estadounidenses Bob Kane y Bill Finger,13​ y propiedad de DC Comics. Apareció por primera vez en la historia titulada «El caso del sindicato químico» de la revista Detective Comics N.º 27, lanzada por la editorial National Publications el 30 de marzo de 1939. La identidad secreta de Batman es Bruce Wayne (Bruno Díaz en algunos países de habla hispana),14​15​16​ un multimillonario magnate empresarial y filántropo dueño de Empresas Wayne en Gotham City. Después de presenciar el asesinato de sus padres, el Dr. Thomas Wayne y Martha Wayne en un violento y fallido asalto cuando era niño, juró venganza contra los criminales, un juramento moderado por el sentido de la justicia."
  },
  {
    id: "dc-superman",
    superhero: "Superman",
    publisher: "DC Comics",
    alter_ego: "Kal-El",
    first_appearance: "Action Comics #1",
    characters: "Kal-El",
    description: "Superman (en español: Superhombre) es un superhéroe ficticio que apareció por primera vez en los cómics estadounidenses publicados por DC Comics.1​2​3​4​El personaje fue creado por el escritor estadounidense Jerry Siegel y el artista canadiense Joe Shuster en 1933 cuando ambos se encontraban viviendo en Cleveland, Ohio. Sus creadores lo vendieron a Detective Comics, Inc. en 1938 por 130 dólares5​ y la primera aventura del personaje fue publicada en Action Comics #1 (junio de 1938), para luego aparecer en varios seriales de radio, programas de televisión, películas, tiras periódicas y videojuegos. Con el éxito de sus aventuras, este personaje definió al superhéroe y estableció su primacía dentro del cómic estadounidense.1​ Su apariencia del personaje es distintiva y se volvió icónica: un traje azul y rojo, con una capa y un emblema con una letra “S” estilizada en su pecho,6​7​8​ el cual se ha convertido en un símbolo del personaje."
  },
  {
    id: "dc-flash",
    superhero: "Flash",
    publisher: "DC Comics",
    alter_ego: "Jay Garrick",
    first_appearance: "Flash Comics #1",
    characters: "Jay Garrick, Barry Allen, Wally West, Bart Allen",
    description: "Flash (conocido también como The Flash y traducido en español: Destello) es el nombre de varios superhéroes ficticios que aparecen en los cómics estadounidenses publicados por DC Comics. Creado por el escritor Gardner Fox y el artista Harry Lampert, el 'Flash' original apareció por primera vez en Flash Comics #1 (fecha de portada de enero de 1940 / mes de noviembre de 1939).1​ Apodado el 'Corredor Escarlata', todas las encarnaciones del Flash poseen 'súper velocidad', que incluye la capacidad de correr, moverse y pensar extremadamente rápido, también puede atravesar la materia sólida, usar reflejos sobrehumanos y aparentemente violar ciertas leyes de la física, como superar la velocidad de la luz."
  },
  {
    id: "dc-green",
    superhero: "Green Lantern",
    publisher: "DC Comics",
    alter_ego: "Alan Scott",
    first_appearance: "All-American Comics #16",
    characters:
      "Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz",
    description:
      "Linterna Verde (en inglés: Green Lantern) es el alias de varios superhéroes de la ficción del Universo DC, de la editorial DC Comics, los cuales se caracterizan por portar un anillo de poder y tener la capacidad de crear manifestaciones de luz sólida con dichos anillos. El primer Linterna Verde (Alan Scott) hizo su debut en julio de 1940 en el All-American Comics #16 (portada de Sheldon Moldoff) durante la Edad de Oro del Cómic estadounidense.1​ Su creación se la debemos al guionista Bill Finger y al dibujante Martín Nodell."
  },
  {
    id: "dc-arrow",
    superhero: "Green Arrow",
    publisher: "DC Comics",
    alter_ego: "Oliver Queen",
    first_appearance: "More Fun Comics #73",
    characters: "Oliver Queen",
    description: 'Green Arrow (en español: Flecha Verde) es un superhéroe que aparece en los cómics publicados por DC Comics. Creado por Mort Weisinger y diseñado por George Papp, apareció por primera vez en More Fun Comics #73 en noviembre de 1941. Su nombre real es Oliver Jonas Queen, un empresario adinerado y propietario de Industrias Queen que también es una celebridad muy conocida en Star City. Utiliza esta posición para ocultar el hecho de que él es la Flecha.1​A veces se muestra vestido como el personaje de Robin Hood, Green Arrow es un arquero que usa sus habilidades para combatir el crimen en sus ciudades natales de Star City y Seattle, así como junto a sus compañeros superhéroes como miembro de la Liga de la Justicia. '
  },
  {
    id: "dc-wonder",
    superhero: "Wonder Woman",
    publisher: "DC Comics",
    alter_ego: "Princess Diana",
    first_appearance: "All Star Comics #8",
    characters: "Princess Diana",
    description: "La Mujer Maravilla (en inglés: Wonder Woman) es una superheroína ficticia creada por William Moulton Marston para la editorial DC Comics. Es una princesa guerrera de las Amazonas, pueblo ficticio basado en el de las amazonas de la mitología griega. En su tierra natal es conocida como la princesa Diana de Temiscira pero fuera de esta utiliza la identidad secreta de Diana Prince. Está dotada de una amplia gama de poderes superhumanos y habilidades de combate de batalla superiores, gracias a sus dones obtenidos de los dioses y su amplio entrenamiento. Ella posee un gran arsenal de armas, incluyendo entre las principales el Lazo de la Verdad, un par de brazaletes mágicos indestructibles."
  },
  {
    id: "dc-martian",
    superhero: "Martian Manhunter",
    publisher: "DC Comics",
    alter_ego: "J'onn J'onzz",
    first_appearance: "Detective Comics #225",
    characters: "Martian Manhunter",
    description: "El Detective Marciano (J'onn J'onzz) (en inglés: Martian Manhunter) es un superhéroe que aparece en los cómics estadounidenses publicados por DC Comics. Creado por el escritor Joseph Samachson y diseñado por el artista Joe Certa, el personaje apareció por primera vez en la historia 'The Manhunter from Mars' en Detective Comics # 225 (noviembre de 1955). Martian Manhunter es uno de los siete miembros originales de la Liga de la Justicia de América y uno de los seres más poderosos del Universo DC."
  },
  {
    id: "dc-robin",
    superhero: "Robin (Nightwing)",
    publisher: "DC Comics",
    alter_ego: "Dick Grayson",
    first_appearance: "Detective Comics #38",
    characters: "Dick Grayson",
    description: "Robin es el alias de varios superhéroes ficticios que aparecen en los cómics estadounidenses publicados por DC Comics. El personaje fue creado originalmente por Bob Kane, Bill Finger y Jerry Robinson, para servir como contraparte menor del superhéroe Batman. La primera encarnación del personaje, Dick Grayson, debutó en Detective Comics # 38 (abril de 1940). Robin hizo apariciones regulares en cómics relacionados con Batman y otras publicaciones de DC Comics desde 1940 hasta principios de la década de 1980 hasta que el personaje dejó de lado la identidad de Robin y se convirtió en el superhéroe independiente Nightwing. El equipo de Batman y Robin se conoce comúnmente como los Caped Crusaders o Dynamic Duo."
  },
  {
    id: "dc-blue",
    superhero: "Blue Beetle",
    publisher: "DC Comics",
    alter_ego: "Dan Garret",
    first_appearance: "Mystery Men Comics #1",
    characters: "Dan Garret, Ted Kord, Jaime Reyes",
    description: "Escarabajo Azul (inglés: Blue Beetle) es el nombre de tres superhéroes ficticios que aparecen en varios cómics estadounidenses publicados por una variedad de compañías desde 1939. La más reciente de las compañías en poseer los derechos del Blue Beetle es DC Comics, que compró los derechos del personaje en 1983, utilizando el nombre de tres personajes distintos a lo largo de los años. El Blue Beetle original fue creado por Fox Comics y luego propiedad de Charlton Comics. El primer Beetle fue Dan Garret (más tarde deletreado Dan Garrett), quien inicialmente obtuvo superpoderes de una vitamina especial, que luego se cambió para obtener poderes de un 'escarabajo sagrado'. El Blue Beetle original apareció no solo en su propio cómic, sino también en una serie de radio semanal."
  },
  {
    id: "dc-black",
    superhero: "Black Canary",
    publisher: "DC Comics",
    alter_ego: "Dinah Drake",
    first_appearance: "Flash Comics #86",
    characters: "Dinah Drake, Dinah Lance",
    description: "Canario Negro (en inglés, Black Canary) es una superheroína ficticia que aparece en los cómics estadounidenses publicados por DC Comics. Creado por el equipo de escritores y artistas de Robert Kanigher y Carmine Infantino, el personaje debutó en Flash Comics # 86 (agosto de 1947). Una de los primeros super-heroínas de DC, Canario Negro ha aparecido en muchas de insignia de la compañía Team-Up títulos como Sociedad de la Justicia de América y la Liga de la Justicia de América. Desde finales de la década de 1960, el personaje ha sido emparejado con el superhéroe arquero Flecha Verde, profesional y románticamente."
  },
  {
    id: "marvel-spider",
    superhero: "Spider Man",
    publisher: "Marvel Comics",
    alter_ego: "Peter Parker",
    first_appearance: "Amazing Fantasy #15",
    characters: "Peter Parker",
    description: "Spider-Man, traducido en ocasiones como el Hombre Araña,11​12​ es un personaje creado por los estadounidenses Stan Lee y Steve Ditko,13​14​ e introducido en el cómic Amazing Fantasy n.° 15, publicado por Marvel Comics en agosto de 1962.15​ Se trata de un superhéroe que emplea sus habilidades sobrehumanas, reminiscentes de una araña, para combatir a otros supervillanos que persiguen fines siniestros. Su creación se remonta a principios de la década de 1960 como respuesta al creciente interés del público adolescente en los cómics y el éxito de Los 4 Fantásticos.16​ Tras su primera aparición en Amazing Fantasy,17​ Marvel decidió producir una serie individual titulada The Amazing Spider-Man, cuyo ejemplar inicial salió a la venta en marzo de 1963."
  },
  {
    id: "marvel-captain",
    superhero: "Captain America",
    publisher: "Marvel Comics",
    alter_ego: "Steve Rogers",
    first_appearance: "Captain America Comics #1",
    characters: "Steve Rogers",
    description: "El Capitán América, cuyo nombre real es Steven 'Steve' Grant Rogers, fue un superhéroe ficticio que aparece en los cómics estadounidenses publicados por Marvel Comics. Creado por los historietistas Joe Simon y Jack Kirby, el personaje apareció por primera vez en Captain America Comics #1 (marzo de 1941) de Timely Comics, predecesor de Marvel Comics. El Capitán América fue diseñado como un supersoldado patriota que luchaba frecuentemente contra las potencias del Eje en la Segunda Guerra Mundial, y fue el personaje más popular de Timely Comics durante el período de guerra."
  },
  {
    id: "marvel-iron",
    superhero: "Iron Man",
    publisher: "Marvel Comics",
    alter_ego: "Tony Stark",
    first_appearance: "Tales of Suspense #39",
    characters: "Tony Stark",
    description: "Iron Man (conocido en español como el Hombre de Hierro) es un superhéroe que aparece en los cómics estadounidenses publicados por Marvel Comics nacido en Sofía, Bulgaria. El personaje fue cocreado por el escritor y editor Stan Lee y el guionista Larry Lieber. Los artistas Don Heck y Jack Kirby fueron los encargados de su diseño. Anthony Edward 'Tony' Stark es un personaje ficticio que da vida a este personaje, un multimillonario magnate empresarial y filántropo estadounidense, playboy e ingenioso científico, que sufrió una grave lesión en el pecho durante un secuestro en el Medio Oriente."
  },
  {
    id: "marvel-thor",
    superhero: "Thor",
    publisher: "Marvel Comics",
    alter_ego: "Thor Odinson",
    first_appearance: "Journey into Myster #83",
    characters: "Thor Odinson",
    description: "Thor es un superhéroe ficticio que aparece en los cómics estadounidenses publicados por Marvel Comics. Es el dios del trueno asgardiano basado en la deidad homónima, del pueblo ficticio inspirado en el mundo de Asgard de la mitología nórdica. En su tierra natal es conocido como Thor Odinson pero fuera de esta utiliza la identidad secreta de Donald Blake.1​ Es poseedor del martillo encantado, Mjolnir, que le otorga capacidad de volar y manipular el clima entre sus otros atributos sobrehumanos, además de concentrar su poder."
  },
  {
    id: "marvel-hulk",
    superhero: "Hulk",
    publisher: "Marvel Comics",
    alter_ego: "Bruce Banner",
    first_appearance: "The Incredible Hulk #1",
    characters: "Bruce Banner",
    description: "Hulk (llamado Hulk o El Hombre Increíble en muchas de las traducciones al español) es un personaje ficticio, un superhéroe que aparece en los cómics estadounidenses publicados por la editorial Marvel Comics, es considerado el personaje más fuerte de Marvel Comics. El personaje fue creado por los escritores Stan Lee y Jack Kirby siendo su primera aparición en The Incredible Hulk #1 publicado en mayo de 1962. En sus apariciones de cómic, el personaje es a la vez Hulk, un ser humanoide enorme de piel verde, corpulento y musculoso que posee una gran fortaleza física, y su alter ego el Dr. Robert Bruce Banner"
  },
  {
    id: "marvel-wolverine",
    superhero: "Wolverine",
    publisher: "Marvel Comics",
    alter_ego: "James Howlett",
    first_appearance: "The Incredible Hulk #180",
    characters: "James Howlett",
    description: "Wolverine, cuyo nombre de nacimiento es James Howlett (también conocido como James Logan o simplemente Logan)3​ es un superhéroe ficticio que aparece en los cómics publicados por Marvel Comics, principalmente en asociación con los X-Men. Es un mutante que posee sentidos afinados a los animales, capacidades físicas mejoradas, poderosa capacidad de regeneración conocida como un factor de curación, y tres garras retráctiles en cada mano. Wolverine ha sido representado de diversas formas como miembro de los X-Men, Alpha Flight, Fuerza-X y Los Vengadores."
  },
  {
    id: "marvel-daredevil",
    superhero: "Daredevil",
    publisher: "Marvel Comics",
    alter_ego: "Matthew Michael Murdock",
    first_appearance: "Daredevil #1",
    characters: "Matthew Michael Murdock",
    description: "Daredevil (también llamado Dan Defensor, Diablo Defensor o Diabólico en muchas de las traducciones al español), alter ego de Matthew Michael 'Matt' Murdock, es un superhéroe ficticio que aparece en los cómics estadounidenses publicados por Marvel Comics. Daredevil fue creado por el escritor y editor Stan Lee y el artista Bill Everett, con una cantidad no especificada de aportes de Jack Kirby. El personaje apareció por primera vez en Daredevil #1 (abril de 1964) en la Edad de Plata de los cómics."
  },
  {
    id: "marvel-hawkeye",
    superhero: "Hawkeye",
    publisher: "Marvel Comics",
    alter_ego: "Clinton Francis Barton",
    first_appearance: "Tales of Suspense #57",
    characters: "Clinton Francis Barton",
    description: "Hawkeye (Clinton Francis «Clint» Barton), es un superhéroe de ficción que aparece en los cómics estadounidenses publicados por Marvel Comics. Creado por el escritor Stan Lee y el artista Don Heck, el personaje apareció por primera vez como un villano en Tales of Suspense # 57 (septiembre de 1964) y más tarde se unió a Los Vengadores en The Avengers # 16 (mayo de 1965). Él ha sido un miembro prominente del equipo desde entonces. También ocupó el puesto # 44 en la lista de los 100 mejores héroes de cómics de IGN"
  },
  {
    id: "marvel-cyclops",
    superhero: "Cyclops",
    publisher: "Marvel Comics",
    alter_ego: "Scott Summers",
    first_appearance: "X-Men #1",
    characters: "Scott Summers",
    description: "Cíclope (Scott Summers) es un personaje ficticio que aparece en los cómics estadounidenses publicados por Marvel Comics y es miembro fundador de los X-Men. Creado por el escritor Stan Lee y el artista Jack Kirby, el personaje apareció por primera vez en el cómic The X-Men de 1963, en la Edad de Plata de los cómics. Cíclope es miembro de una subespecie de humanos conocidos como mutantes, que nacen con habilidades sobrehumanas. Cíclope emite poderosos rayos de energía de sus ojos y solo puede controlar los rayos con la ayuda de gafas especiales que debe usar en todo momento. "
  },
  {
    id: "marvel-silver",
    superhero: "Silver Surfer",
    publisher: "Marvel Comics",
    alter_ego: "Norrin Radd",
    first_appearance: "The Fantastic Four #48",
    characters: "Norrin Radd",
    description: "Silver Surfer (también conocido en Hispanoamérica como Surfista de Plata, Surfista Plateado o Deslizador de plata, y en España como Estela Plateada) es un personaje ficticio de cómic del Universo Marvel creado por Jack Kirby. Su primera aparición fue en Los 4 Fantásticos n.° 48 (marzo de 1966). El personaje también aparece en adaptaciones al cine, la televisión y los videojuegos. El Silver Surfer es un extraterrestre humanoide con piel metálica que puede viajar a través del espacio con la ayuda de su nave, similar a una tabla de surf. Originalmente un joven astrónomo de nombre Norrin Radd en el planeta Zenn-La, salvó a su planeta de Galactus, el devorador de planetas, sirviendo como su heraldo."
  },
];
