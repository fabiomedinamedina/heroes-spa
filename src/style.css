/* GENERALS */
:root {
  --green: #38EF7D;
  --blue-dark: #0A2E36;
}

html, body {
  font-family: 'Poppins', sans-serif;
  font-weight: 400;
  background-image: url('/assets/background-pages.jpg');
  background-attachment: fixed;
  background-position: center;
  background-size: cover;
}

.container-page > .container {
  margin-top: 8rem;
}

.text-color-secondary {
  color: #3F5E5A !important;
}

/* NAVBAR HEADER */
.navbar {
  position: fixed;
  top: 1rem;
  left: 1.5rem;
  width: calc(100% - 3rem);
  z-index: 99;
  padding: 0.5rem 1rem;
}

@media (min-width: 768px) {
  .navbar {
    top: 1.5rem;
    left: 2.5rem;
    width: calc(100% - 5rem);
    padding: 0.5rem 2rem;
  }
}

.bg-navbar{
  background: rgba(10, 46, 54, 0.8);
  backdrop-filter: blur(2px);
}

.navbar .navbar-brand {
  font-weight: 700;
}

.navbar .navbar-brand small{
  font-size: 0.8rem;
  font-weight: 400;
}

.navbar .nav-link {
  color: #FFF;
}

.navbar .nav-link.active {
  color: var(--green);
  text-decoration: underline;
}

.navbar-toggler {
  border: 1px solid var(--green);
  color: var(--green);
}

.navbar .name-user strong{
  font-weight: 700;
  color: var(--green);
}

/* CARDS PUBLISHER AND SEARCH */

.card-hero {
  position: relative;
  box-shadow: 0px 10px 25px rgba(0, 0, 0, 0.1), 0px 4px 4px rgba(0, 0, 0, 0.1);;
}

.info-publisher {
  z-index: 9;
}

.info-publisher img { 
  max-width: 180px;
}

.info-publisher h1 {
  font-size: 1.7rem;
  font-weight: 400;
  color: #FFF;
}

.content-hero-card:after {
  content: '';
  width: 100%;
  height: 200px;
  position: absolute;
  bottom: 0;
  left: 0;
  background: linear-gradient(180deg, rgba(0, 0, 0, 0) 40%, #000000 104%);
  border-radius: 0 0 1rem 1rem;
  transition: all 0.2s ease-in-out;
}

.content-publisher {
  position: relative;
}

.content-publisher:after {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  background: linear-gradient(90deg, rgba(0, 0, 0, 0) 0%, #000000 100%);
  border-radius: 0  1rem  1rem 0;
  opacity: 0.8;
}

@media (min-width: 767px) {
  .content-publisher:after {
    width: 300px;
    background: linear-gradient(90deg, rgba(0, 0, 0, 0) 40%, #000000 104%);
  }
}

.link-hero:hover .content-hero-card:after {
  height: 250px;
}

.info-hero {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  z-index: 9;
  transition: all 0.2s ease-in-out;
}

@media (min-width: 768px) {
  .link-hero:hover .info-hero{
    bottom: 1rem;
  }
}

.info-hero h2 {
  color: #FFF;
  font-weight: 700;
  font-size: 1.2rem;
}

@media (min-width: 768px) {
  .info-hero h2 {
    font-size: 1.5rem;
  }
}

.info-hero p,
.content-search p {
  color: #FFF;
}

.content-hero-card .badge {
  position: absolute;
  right: 1rem;
  top: 1rem;
}

.badge-dc-comics{
  background-color: #0376F2;
}

.badge-marvel-comics{
  background-color: #EB1D25;
  ;
}

/* SEARCH PAGE */

.content-search h1 {
  font-weight: 700;
}

.form-search input {
  z-index: 2;
  position: relative;
  border: 1px solid #FFF;
  background-color: transparent;
  color: #FFF;
}

@media (min-width: 768px) {
  .form-search input {
    transform: translateX(4rem);
    padding-right: 4rem;
  }
}

.form-search input:focus {
  background-color: transparent;
  color: #FFF;
  border: 1px solid var(--green);
}

.form-search input::placeholder{
  color: #FFF;
}

.form-search button,
.btn-login {
  z-index: 9;
  position: relative;
  border: 1px solid var(--green);
  background-color: var(--green);
  font-weight: 700;
  color: #FFF;
}

.form-search button:hover,
.btn-login:hover,
.form-search button:first-child:active,
.btn-login:first-child:active {
  background-color: #29d76c;
  border: 1px solid #29d76c;
  color: #FFF;
}

.not-found-heros h2,
.not-found-heros p,
.login-row h1,
.login-row p,
.card-hero-single h1,
.card-hero-single h2,
.card-hero-single p,
.card-hero-single {
  color: var(--blue-dark);
}

/* LOGIN PAGE */

@media (min-width: 767px) {
  .login-row {
    height: 100vh;
  }
}

.logo-fm {
  max-width: 200px;
}

.left-login {
  padding: 3rem;
}

@media (min-width: 768px) {
  .left-login {
    padding: 6rem 7rem;
  }
}

.login-form-content h1 {
  font-weight: 700;
}

.right-login {
  background: url('assets/back-login.jpg');
}

.right-login img {
  max-width: 80%;
  margin-inline: auto;
}

.form-login input {
  border: 1px solid #E5E5E5;
  background-color: transparent;
  color: #ADADAD;
}

.form-login input:focus {
  background-color: transparent;
  color: #ADADAD;
  border: 1px solid var(--green);
}

.form-login input::placeholder{
  color: #ADADAD;
}

.legal-login {
  font-size: 0.8rem;
  font-style: italic;
}

/* SINGLE HERO PAGE */
.card-hero-single img {
  border-radius: 1rem 1rem 0 0 ;
}

@media (min-width: 768px) {
  .card-hero-single img {
    border-radius: 1rem 0 0 1rem;
  }
}

.card-hero-single .card-body {
  padding: 2.5rem;
}

@media (min-width: 768px) {
  .card-hero-single .card-body {
    padding: 3.5rem 6rem;
  }
}

/* FOOTER */

.copyright-link {
  font-weight: 700;
  color: var(--green);
  transition: all 0.2s ease-in-out;
}

.copyright-link:is(:active, :hover) {
  color: #29d76c;
}

.social-media {
  gap: 0.5rem;
}

.social-media a{
  font-size: 0.8rem;
}

.social-media a:not(:last-child) {
  padding-right: 1rem;
}