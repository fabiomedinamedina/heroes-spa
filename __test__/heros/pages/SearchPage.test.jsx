import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heros';


const mockUseNavigate =  jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUseNavigate
}));


describe('Pruebas en <SearchPage />', () => {

  beforeEach( () => jest.clearAllMocks() );
  
  test('Debería mostrarse correctamente con valores por defecto', () => {
    
    const {container}  = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect( container ).toMatchSnapshot();
    expect( screen.getByRole( 'button' ) ).toBeTruthy();


  });

  test('Debería mostrar a batman y el input con el valor del queryString', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    );

    expect( screen.getByRole( 'textbox' ).value ).toBe( 'batman' );
    expect( screen.getByLabelText( "img-hero" ).src ).toContain('/assets/heroes/dc-batman.jpg');

  });

  test('Debería mostrar un error si el hero no existe', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman22']}>
        <SearchPage />
      </MemoryRouter>
    );

    // screen.debug();
    expect( screen.getByRole( 'heading', { level: 2 } ).innerHTML ).toContain( '¡Disculpa!' );
    expect( screen.getByText( 'No se encuntraron héroes con XXXX, te pedimos que lo intentes con otro nombre.' ).innerHTML ).toBeTruthy();
  });

  test('Debería llamar el navigate a la pantalla nueva', () => {

    const stringSearch = 'Superman';

    render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    const inputText = screen.getByRole( 'textbox' );
    const buttonSearch = screen.getByRole( 'button' );
    fireEvent.change( inputText, {target: {value: stringSearch}} );
    fireEvent.click( buttonSearch );
    
    expect( mockUseNavigate ).toHaveBeenCalled();
    expect( mockUseNavigate ).toHaveBeenCalledWith( `?q=${stringSearch}` );
    
  });

})