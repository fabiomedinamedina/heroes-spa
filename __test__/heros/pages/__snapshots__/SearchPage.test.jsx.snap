// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> Debería mostrarse correctamente con valores por defecto 1`] = `
<div>
  <div
    class="container container-title-publisher  animate__animated animate__fadeIn"
  >
    <div
      class="content-search p-4 p-lg-5 d-flex justify-content-between align-items-end rounded-4 mb-5"
      style="background-image: url(assets/titles/background-search.jpg); background-repeat: no-repeat; background-size: cover;"
    >
      <div
        class="container"
      >
        <div
          class="row align-items-end"
        >
          <div
            class="col-12 col-lg-6"
          >
            <div
              class="info-publisher"
            >
              <img
                alt="Logos DC y Marvel"
                class="img-fluid pb-2"
                src="assets/titles/logos-dc-marvel.svg"
              />
              <h1>
                Buscador de héroes
              </h1>
              <p
                class="mb-0"
              >
                Escribe el nombre del héroe que quieres buscar. Recuerda que debes escribir más de 3 caracteres
              </p>
            </div>
          </div>
          <div
            class="col-12 col-lg-6"
          >
            <form
              class="form-search d-flex flex-column flex-lg-row justify-content-end flex-column pb-2 pt-3 pt-lg-0"
            >
              <div
                class="col-auto mt-0"
              >
                <input
                  autocomplete="off"
                  class="form-control rounded-pill"
                  name="searchText"
                  placeholder="Escribe el nombre de un héroe."
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-auto mt-0 mb-0"
              >
                <button
                  class="btn rounded-pill px-4 mt-2 mt-lg-0"
                  type="submit"
                >
                  Buscar Héroes
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="row results-heros animate__animated animate__fadeIn g-3"
    />
  </div>
</div>
`;
